documentation/api/addAssertion.md

```diff
@@ -10,82 +10,208 @@ expect.addAssertion(pattern, handler);
-  '<array> [not] to be (sorted|ordered) <function?>',
-  function(expect, subject, cmp) {
-    expect.errorMode = errorMode;
-    expect(subject, '[not] to equal', [].concat(subject).sort(cmp));
+  '<array> to have item <any>',
+  function(expect, subject, value) {
+    expect(subject, 'to contain', value);
   }
 );
 ```
 
-The above assertion definition makes the following expects possible:
+A handler function can use other assertions, including other custom assertions
+previously added via `expect.addAssertion`. This way, one could build up complex assertions from simpler ones or just reword an existing assertion, like in this example.
+
+The new assertion can then be used as follows:
 
 ```js
-expect([1, 2, 3], 'to be sorted');
-expect([1, 2, 3], 'to be ordered');
-expect([2, 1, 3], 'not to be sorted');
-expect([2, 1, 3], 'not to be ordered');
-expect([3, 2, 1], 'to be sorted', function(x, y) {
-  return y - x;
-});
+expect([1, 2, 3], 'to have item', 2);
 ```
 
-Let's dissect the different parts of the custom assertion we just
-introduced.
-
 The first parameter to `addAssertion` is a string or an array of strings
-stating the patterns this assertion should match. A pattern has the following
-syntax. A word in angle brackets represents a type of either the subject
-or one of the parameters to the assertion. In this case the assertion
-is only defined for arrays. If no subject type is specified,
-the assertion will be defined for the type `any`, and would
-be applicable any type. See the `Extending Unexpected with new types`
-section for more information about the type system in Unexpected.
-
-A word in square brackets represents a flag that can either be
-there or not. If the flag is present `expect.flags[flag]` will contain
-the value `true`. In this case `not` is a flag. When a flag it present
-in a nested `expect` it will be inserted if the flag is present;
-otherwise it will be removed. Text that is in parentheses with
-vertical bars between them are treated as alternative texts that can
-be used. In this case you can write _ordered_ as an alternative to
-_sorted_.
+describing the pattern(s) the assertion should match. The pattern takes the
+following structure:
+
+`<the-subject> an assertion string <the-value>`
+
+The words in angle brackets define what types the assertion applies to. These
+can be any of the internally-defined types or new types added via
+[expect.addType](../addType). In this example, the subject to `to have item`
+must be an array, while the value may be of any type.
+
+If mismatching types are used, Unexpected throws an error with a helpful
+suggestion:
+
+```js
+expect('abcd', 'to have item', 'a');
+```
+
+```output
+expect('abcd', 'to have item', 'a');
+The assertion does not have a matching signature for:
+    <string> to have item <string>
+  did you mean:
+    <array> to have item <any>
+```
+
+If no subject type is specified, the assertion will be defined for the
+type `any` and would be applicable to any type.
```

Comment:
I think we should remove this sentence, as omitting all types [results in a deprecation warning since 11.1.0](https://github.com/unexpectedjs/unexpected/blob/master/CHANGELOG.md#v1110-2019-02-25), and omitting the subject type results in (for example) `SyntaxError: Missing subject type in to be foo <string>`.